tool/plugin/lib/plugin.dart

```diff
@@ -160,7 +177,7 @@ class BuildCommand extends ProductCommand {
 
   Future<int> doit() async {
```

Comment:
General question: do we need to be async?  What bits will take advantage of running asynchronously?  (Sorry if this is obvious!)

Reply:
These methods run a lot of shell commands. Like curl, javac, jar, cp, git, etc. We need to use async to keep it all in order.